CRM/Core/Payment.php

```diff
@@ -1060,27 +1225,48 @@ protected function getBaseReturnUrl() {
   /**
    * Get the currency for the transaction.
    *
-   * Handle any inconsistency about how it is passed in here.
+   * Legacy wrapper. Better for a method to work on its own PropertyBag.
+   *
+   * When we introduced propertyBag this code was adjusted to try to preserve
+   * the previous behaviour.
    *
    * @param $params
    *
    * @return string
    */
-  protected function getCurrency($params) {
-    return CRM_Utils_Array::value('currencyID', $params, CRM_Utils_Array::value('currency', $params));
+  protected function getCurrency($params = []) {
```

Comment:
Ah you still have it ... do we need it above?

Reply:
Yeah I think outside core there might be processors that do. You can throw a deprecation notice into it once we get PayPaylImpl  not using it